packages/wallet/src/redux/indirect-funding/player-a/reducer.ts

```diff
@@ -18,19 +39,365 @@ export function playerAReducer(
 
   switch (state.indirectFunding.type) {
     case states.WAIT_FOR_APPROVAL:
+      return waitForApprovalReducer(state, action);
     case states.WAIT_FOR_PRE_FUND_SETUP_1:
-    // pass the commitment to the channel state reducer
-    // update channel state based on outcome
-    // if ready to fund:
-    //   initiate direct funding
+      return waitForPreFundSetup1Reducer(state, action);
     case states.WAIT_FOR_DIRECT_FUNDING:
-    // progress direct funding
-    // if direct funding is finished:
-    //   send and/or wait for post-fund setup
+      return waitForDirectFunding(state, action);
     case states.WAIT_FOR_POST_FUND_SETUP_1:
+      return waitForPostFundSetup1(state, action);
     case states.WAIT_FOR_LEDGER_UPDATE_1:
-      return state;
+      return waitForLedgerUpdateReducer(state, action);
     default:
       return unreachable(state.indirectFunding);
   }
 }
+
+const waitForLedgerUpdateReducer = (
+  state: walletStates.Initialized,
+  action: actions.indirectFunding.Action,
+): walletStates.Initialized => {
+  switch (action.type) {
+    case actions.COMMITMENT_RECEIVED:
+      const indirectFundingState = selectors.getIndirectFundingState(
+        state,
+      ) as states.WaitForLedgerUpdate1;
+
+      // Update ledger state
+      let newState = receiveLedgerCommitment(state, action.commitment, action.signature);
+      if (
+        ledgerChannelFundsAppChannel(
+          newState,
+          indirectFundingState.channelId,
+          indirectFundingState.ledgerId,
+        )
+      ) {
+        newState = confirmFundingForAppChannel(newState, indirectFundingState.channelId);
+      }
+      return newState;
+    default:
+      return state;
+  }
+};
+
+const waitForPostFundSetup1 = (
+  state: walletStates.Initialized,
+  action: actions.indirectFunding.Action,
+): walletStates.Initialized => {
+  switch (action.type) {
+    case actions.COMMITMENT_RECEIVED:
+      const indirectFundingState = selectors.getIndirectFundingState(
+        state,
+      ) as states.WaitForPostFundSetup1;
```

Comment:
Is this type assertion even necessary anymore? I am not using it in the player-b reducer.

Reply:
The only time we need to do this is when we want to pull `ledgerId` off of `indirectFundingState` since it's not on all `indirectFundingState`.